4.0.0/migration.md

```diff
@@ -19,6 +20,124 @@ The migration process is initiated by the Installer if the following requirement
+                `TABLE_NAME` VARCHAR(64), 
+                `COLUMN_NAME` VARCHAR(64), 
+                `CONSTRAINT_NAME` VARCHAR(64),
+                `REFERENCED_TABLE_SCHEMA` VARCHAR(64),
+                `REFERENCED_TABLE_NAME` VARCHAR(64),
+                `REFERENCED_COLUMN_NAME` VARCHAR(64),
+                `INVALID_KEY_COUNT` INT,
+                `INVALID_KEY_SQL` VARCHAR(1024)
+            );
+        END IF;
+        OPEN foreign_key_cursor;
+        foreign_key_cursor_loop: LOOP
+            FETCH foreign_key_cursor INTO 
+            TABLE_SCHEMA_VAR, 
+            TABLE_NAME_VAR, 
+            COLUMN_NAME_VAR, 
+            CONSTRAINT_NAME_VAR, 
+            REFERENCED_TABLE_SCHEMA_VAR, 
+            REFERENCED_TABLE_NAME_VAR, 
+            REFERENCED_COLUMN_NAME_VAR;
+            IF done THEN
+                LEAVE foreign_key_cursor_loop;
+            END IF;
+            SET @from_part = CONCAT('FROM ', '`', TABLE_SCHEMA_VAR, '`.`', TABLE_NAME_VAR, '`', ' AS REFERRING ', 
+                 'LEFT JOIN `', REFERENCED_TABLE_SCHEMA_VAR, '`.`', REFERENCED_TABLE_NAME_VAR, '`', ' AS REFERRED ', 
+                 'ON (REFERRING', '.`', COLUMN_NAME_VAR, '`', ' = ', 'REFERRED', '.`', REFERENCED_COLUMN_NAME_VAR, '`', ') ', 
+                 'WHERE REFERRING', '.`', COLUMN_NAME_VAR, '`', ' IS NOT NULL ',
+                 'AND REFERRED', '.`', REFERENCED_COLUMN_NAME_VAR, '`', ' IS NULL');
+            SET @full_query = CONCAT('SELECT COUNT(*) ', @from_part, ' INTO @invalid_key_count;');
+            PREPARE stmt FROM @full_query;
+            EXECUTE stmt;
+            IF @invalid_key_count > 0 THEN
+                INSERT INTO 
+                    INVALID_FOREIGN_KEYS 
+                SET 
+                    `TABLE_SCHEMA` = TABLE_SCHEMA_VAR, 
+                    `TABLE_NAME` = TABLE_NAME_VAR, 
+                    `COLUMN_NAME` = COLUMN_NAME_VAR, 
+                    `CONSTRAINT_NAME` = CONSTRAINT_NAME_VAR, 
+                    `REFERENCED_TABLE_SCHEMA` = REFERENCED_TABLE_SCHEMA_VAR, 
+                    `REFERENCED_TABLE_NAME` = REFERENCED_TABLE_NAME_VAR, 
+                    `REFERENCED_COLUMN_NAME` = REFERENCED_COLUMN_NAME_VAR, 
+                    `INVALID_KEY_COUNT` = @invalid_key_count,
+                    `INVALID_KEY_SQL` = CONCAT('SELECT ', 
+                        'REFERRING.', '`', COLUMN_NAME_VAR, '` ', 'AS "Invalid: ', COLUMN_NAME_VAR, '", ', 
+                        'REFERRING.* ', 
+                        @from_part, ';');
+            END IF;
+            DEALLOCATE PREPARE stmt; 
+        END LOOP foreign_key_cursor_loop;
+    END$$
+DELIMITER ;
+CALL ANALYZE_INVALID_FOREIGN_KEYS('%', '%', 'Y');
+DROP PROCEDURE IF EXISTS ANALYZE_INVALID_FOREIGN_KEYS;
+SELECT * FROM INVALID_FOREIGN_KEYS;
+```
+
+If the query does not return any row, the migration process can be initiated. Otherwise, it will be needed to delete the orphaned rows before starting with the migration.
```

Comment:
If the query does not return any row, initiate the migration process. Otherwise, delete the orphaned rows before starting with the migration.